lib/validator/v3/validator.go

```diff
@@ -695,6 +696,11 @@ func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 		}
 	}
 
+	if rule.ICMP != nil && (rule.Protocol == nil || (*rule.Protocol != numorstring.ProtocolFromString("ICMP") &&
```

Comment:
I think Felix could be tripped up by having a rule with `ipVersion=4` and `protocol=ICMPv6` too (or vice versa).  Perhaps we need to validate against that here too, or adjust the logic in felix so that it skips such inconsistent rules.

Reply:
A rule with mismatched ipversion and protocol is a useless rule but I don't think its really an "invalid" rule. A good example is the inverse - should we reject a rule if it states `ipVersion=6 && !protocol=icmp`? The !protocol is useless but not necessarily invalid.

Reply:
I mean, I think that's another thing that'd crash felix right now: trying to render an IPv4 rule with ICMPv6 or vice-versa.  We should fix it here by validating against it or in Felix (by ignoring the rule on the basis that it can't possibly match anything because it specified conflicting match criteria).

Reply:
I _think_ validating against it here is better because it's probably a bug in the user's rule so calicoctl will give them better diags than a silent ignore in Felix.

Reply:
addressed. PTAL @fasaxc